docs/pages/developers/evm/filler.mdx

```diff
@@ -1,272 +1,786 @@
-### IntentFiller Class
+Costs:
+- Fill Gas: ~$0.10 (varies by chain)
+- Protocol Fee: ~$0.95 (per byte fee * message size)
+- Relayer Fee: $0.25 (fixed)
+- Total Costs: ~$1.30
+
+Profit: $2.80 - $1.30 = $1.50
+```
+
+### Risk Factors
+
+- **Gas Price Volatility**: High gas prices can eliminate profits
+- **Order Competition**: Other fillers might fill orders first
+- **Token Price Movements**: Prices can change between evaluation and execution
+- **Slippage**: DEX trades might get worse prices than expected
+
+## Custom Strategy
+
+You can implement custom strategies for specialized use cases:
 
-#### Constructor
 ```typescript
-constructor(
-  chainConfigs: ChainConfig[],
-  strategies: FillerStrategy[],
-  config: FillerConfig
-)
+import { FillerStrategy, Order, ExecutionResult } from "@hyperbridge/filler"
+
+class MyArbitrageStrategy implements FillerStrategy {
+	name = "ArbitrageStrategy"
+
+	async canFill(order: Order): Promise<boolean> {
+		// Check if arbitrage opportunity exists
+		const priceA = await this.getTokenPrice(order.inputs[0].token, order.sourceChain)
+		const priceB = await this.getTokenPrice(order.outputs[0].token, order.destChain)
+		return priceA < priceB * 0.95 // 5% arbitrage threshold
+	}
+
+	async calculateProfitability(order: Order): Promise<bigint> {
+		// Calculate arbitrage profit + order fees - costs
+		const arbitrageProfit = await this.calculateArbitrage(order)
+		const orderFees = order.fees
+		const costs = await this.estimateCosts(order)
+		return arbitrageProfit + orderFees - costs
+	}
+
+	async executeOrder(order: Order): Promise<ExecutionResult> {
+		// Execute complex arbitrage + fill strategy
+		return await this.executeArbitrageAndFill(order)
+	}
+}
+```
+
+## Configuration
+
+### Confirmation Policy - Security vs Speed
+
+The confirmation policy is a **critical security feature** that determines how many blockchain confirmations to wait for before processing an order. This prevents attacks from blockchain reorganizations (reorgs) where transactions can be reversed.
+
+#### How It Works
+
+The filler uses a **sliding scale** based on order value:
+
+- **Low-value orders**: Fewer confirmations (faster processing)
+- **High-value orders**: More confirmations (higher security)
+
+```
+Confirmations = Linear interpolation between min/max based on order value
+```
+
+#### Configuration Format
+
+```toml
+[confirmationPolicies."1"]  # Ethereum mainnet
+minAmount = "1000000000000000000"      # 1 ETH
+maxAmount = "100000000000000000000"    # 100 ETH
+minConfirmations = 12                  # 12 blocks for small orders (~3 minutes)
+maxConfirmations = 30                  # 30 blocks for large orders (~7.5 minutes)
+
+[confirmationPolicies."97"]  # BSC testnet
+minAmount = "1000000000000000000"      # 1 BNB
+maxAmount = "10000000000000000000"     # 10 BNB
+minConfirmations = 1                   # 1 block for small orders (~3 seconds)
+maxConfirmations = 5                   # 5 blocks for large orders (~15 seconds)
```

Comment:
Thought this was in USD? might be a good idea that these values are not in 18 decimals now that i think about it. Since fee token can have different decimals

Reply:
When we get the order inputs USD value, we get them in 18 decimals (irrespective of the order input token[i] decimal). So keep these confirmation policy usd value in 18 decimals checks out. 

I think I can simplify it further with just simple usd values without decimals. 

And then we can remove the decimal from the confirmation policy as well 👍🏻